src/command_manager.cc

```diff
@@ -132,7 +132,7 @@ ParseResult parse_quoted(ParseState& state, Delimiter delimiter)
         if (c == delimiter)
         {
             auto next = state.pos;
-            if (read(next, end) != delimiter)
+            if (next == end || read(next, end) != delimiter)
```

Comment:
Often, the last read() will return the terminating zero-byte, but that breaks if the input is not zero-terminated.
I wonder when that happens. Anyway, looks good.